src/charm.py

```diff
@@ -35,6 +43,23 @@
 ]
 
 
+class SpeakerConfig(BaseModel):
+    name: str = Field(
+        ...,
+        regex="^[a-z0-9.-]+$",
+    )
+    node_selector: str = Field(
+        ...,
+        alias="node-selector",
+        regex="^[a-zA-Z0-9A-Za-z0-9_./-]+=[a-zA-Z0-9_.-]+$",
```

Comment:
this regex is mighty big: could you do with this?

```suggestion
        regex="^[\w./-]+=[\w.-]+$",
```

`^` -- starting at the beginning of the line
`[` Any character in this set
* `\w` -- alphanumeric and underscore
* `.` -- period
* `/` -- forward slash
* `-` -- minus
`]+` at least once or more

`=` followed by an equals

`[` Any character in this set
* `\w` -- alphanumeric and underscore
* `.` -- period
* `-` -- minus
`]+` at least once or more
`$` -- ending the line with the previous set